lib/elixir/lib/kernel.ex

```diff
@@ -3669,48 +3676,117 @@ defmodule Kernel do
     end
   end
 
-  defp in_list(left, head, tail, expand, right, in_module?) do
-    fun = fn elem, acc ->
+  defp in_list(left, {:|, _, [head, tail]}, [], expand, right, in_module?, _use_right?, acc) do
```

Comment:
This is an optimization: new argument `use_right?` keeps track of when `right` can be used, so we don't need to reverse `acc` and unquote `right` directly

Reply:
> keeps track of when right can be used

What do you mean by "use"? :)